docs/troubleshooting.asciidoc

```diff
@@ -28,3 +28,72 @@ include::copied-from-beats/getting-help.asciidoc[]
 == Debug
 
 include::copied-from-beats/debugging.asciidoc[]
+
+[[common-problems]]
+== Common problems
+[[queue-is-full]]
+=== 503: Queue is full
+APM Server has an internal queue which stores documents until they can be delivered to Elasticsearch.
+It is designed this way to alleviate problems that might occur if Elasticsearch is intermittently unavailable or if a large spike of data arrives at the APM Server at the same time.
+
+When the internal queue is full,
+APM Server will return a HTTP 503 status with the message "Queue is full".
+
+A full queue generally means that the amount of data being collected by agents and sent to APM Server is greater than the amount of data that the Elasticsearch cluster can ingest.
```

Comment:
How about `than the amount of data that can be processed by the APM Server and ingested by the ES cluster` ?

Otherwise people might only start to fine tune the ES cluster, but it could also be that they need to set up additional APM Server

Reply:
Until recently, increasing APM Server processing power would never fix "Queue is full". Maybe that changed with the "early return" PR?

Reply:
If you have multiple APM Server instances running behind a load balancer adding additional machines basically increases the number of concurrent_requests to be processed and also the internal memory queue, as you have more instances running that handle incoming requests.

Reply:
the general point i was trying to make in these two is:

Queue is full: Elasticsearch is the bottleneck
Request timed out waiting to be processed: APM Server is the bottleneck

I'll try to make this clearer